spec/forms/inheritance_form_spec.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+RSpec.describe AdminForm do
+  context 'class methods' do
+    describe '.properties' do
+      it 'returns an array with email and role' do
+        expect(AdminForm.properties).to contain_exactly(:email, :role)
+      end
+    end
+  end
+
+  context 'instance methods' do
+    let(:user) { build :user }
+    subject { described_class.new(user) }
+
+    describe 'accessors for properties' do
+      it 'allows reading and writing for email' do
+        subject.email = 'admin@example.com'
+
+        expect(subject.email).to eq 'admin@example.com'
+      end
+
+      it 'allows reading and writing for role' do
+        subject.role = 'superadmin'
+
+        expect(subject.role).to eq 'superadmin'
+      end
+    end
+
+    describe 'accessors for AdminForm specific properties' do
+      before do
+        AdminForm.properties :permissions
```

Comment:
It seems like you can add `:permissions` property directly to AdminForm file (spec/dummy/app/forms/admin_form.rb), as you rely on its content anyway

Reply:
I test `AdminForm` without its permissions in previous tests. Using of `Admin.properties` is just easy-way to test this stuff.

Reply:
Yes, I see, the point is that you can add :permissions to the file and your tests will test the same things

Reply:
Fixed :slightly_smiling_face: